cpp/src/arrow/flight/type_fwd.h

```diff
@@ -55,11 +51,46 @@ class ServerCallContext;
 class ServerMiddleware;
 class ServerMiddlewareFactory;
 struct Ticket;
-namespace internal {
+}  // namespace arrow::flight
+
+namespace arrow::flight::internal {
 class AsyncRpc;
 class ClientTransport;
-struct FlightData;
 class ServerTransport;
-}  // namespace internal
-}  // namespace flight
-}  // namespace arrow
+struct FlightData;
+}  // namespace arrow::flight::internal
+
+// Protobuf types from Flight.proto
+namespace arrow::flight::protocol {
+class Action;
+class ActionType;
+class BasicAuth;
+class CancelFlightInfoRequest;
+class CancelFlightInfoResult;
+class Criteria;
+class FlightData;
+class FlightDescriptor;
+class FlightEndpoint;
+class FlightInfo;
+class GetSessionOptionsRequest;
+class Location;
+class PollInfo;
+class RenewFlightEndpointRequest;
+class Result;
+class SchemaResult;
+class SetSessionOptionsRequest;
+class SetSessionOptionsResult;
+class Ticket;
+class GetSessionOptionsRequest;
+class GetSessionOptionsResult;
+class CloseSessionRequest;
+class CloseSessionResult;
+}  // namespace arrow::flight::protocol
+
+namespace arrow::internal {
+class Uri;
+}
+
+namespace google::protobuf {
```

Comment:
Same here. There shouldn't be any protobuf headers in our headers...

Reply:
Right. I will move these to an internal header.

Me next goal is being able to compile flight without linking to protobuf until the user actually links against `libarrow_flight` and the `libprotobuf` version of their choice.